core/tests/protractor_mobile/playerFlow.js

```diff
@@ -0,0 +1,73 @@
+// Copyright 2018 The Oppia Authors. All Rights Reserved.
```

Comment:
Why would the player flow be different for mobile and desktop? The more things we can (conceptually) put into protractor/, the better.

Reply:
The desktop's editorAndPlayer e2e uses exp that is created during run time. Therefore editorAndPlayer's exploration is not usable on mobile tests.

One alternative I can think of is setup a bunch of if() statement, that looks for DEV_MODE, to trigger these learnerFlow tests and incorporate with current editorAndPlayer.

What do you think @abp7?

Reply:
I think I am with @abp7. There is no easy way of running editorAndPlayer in its entirety on mobile tests.

We can detect for DEV_MODE in editorAndPlayer and only run tests without DEV_MODE. But that would be more confusing imo.

Reply:
If the issue is only in setup of explorations, see my comment above: https://github.com/oppia/oppia/pull/5340#discussion_r205960068 . Post-setup, the tests could be the same. The existing tests could also be restructured if necessary (especially if they cover duplicate things).

If the issue is more widespread than that then yeah, maybe fully separating might be a good idea.

Reply:
Here, I think we might be a bit better off keeping them separate since the `beforeAll` blocks completely differ and might result in duplication of the blocks if we try to bring them together. That said, I will surely take a look and make necessary changes.

Reply:
This has been kept as-is since there seems to be almost no duplicacy in the specs.